docker/Dockerfile.rust.sgx

```diff
@@ -0,0 +1,33 @@
+From ubuntu:18.04
+LABEL maintainer="linfeng@crypto.com"
+
+
+RUN apt-get update -y && \
+    apt-get install -y \
+    curl \
+    cmake \
+    libgflags-dev \
+    libzmq3-dev \
+    pkg-config \
+    clang \
+    autoconf \
+    wget \
+    git \
+    libtool \
+    && rm -rf /var/lib/apt/lists/*
+
+
+ENV RUSTUP_HOME=/usr/local/rustup  \
+    CARGO_HOME=/usr/local/cargo  \
+    PATH=/usr/local/cargo/bin:$PATH
+
+RUN set -eux; curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- --default-toolchain nightly-2019-11-25-x86_64-unknown-linux-gnu -y
+
+RUN mkdir -p /opt/intel && \
+    cd /opt/intel && \
+    set -eux; \
+    curl -sSf https://download.01.org/intel-sgx/sgx-linux/2.7.1/distro/ubuntu18.04-server/sgx_linux_x64_sdk_2.7.101.3.bin > sdk.bin &&\
+    chmod +x sdk.bin &&\
+    echo yes | ./sdk.bin &&\
+    rm sdk.bin
```

Comment:
does this make sense? the driver needs to be installed in the host, not in the guest OS

Reply:
the file `Dockerfile.rust.sgx` is used to compile the  `enclave-query` and `enclave-validation` in docker, when we compile the enclave, we need the sgx sdk.

* First, we use this docker file to build an image like  `rust-sgx`
* Then we can map the directory `chain` into the docker container using `-v pwd:/chain`, then run `make` in the container to compile the `enclave-query` and `enclave-validation` , the binary files and `*.so` files are still in the host's chain directory, so we can copy the binary file to a ubuntu host or use these files to build the `crypto-sgx` docker image.